tests/unit/providers/test_vcluster.py

```
@@ -0,0 +1,142 @@
+
+def test_resolve_variables_host_cloud_env(
+    ctx: ResolveContext, monkeypatch: pytest.MonkeyPatch
+) -> None:
+    monkeypatch.setenv("VCLUSTER_HOST_CLOUD", "eks")
+    variables = VclusterProvider().resolve_variables(ctx, {})
+    assert variables["host_cloud"] == "eks"
+
+
+def test_resolve_variables_custom_precedence(ctx: ResolveContext) -> None:
+    variables = VclusterProvider().resolve_variables(
+        ctx, {"cluster_name": "override", "kubeconfig_path": "/tmp/custom.yaml"}
```

Ruff (0.16.1)

[error] Probable insecure usage of temporary file or directory: "/tmp/custom.yaml"

(S108)